### DISCUSSION TITLE
Request: Independent Parser Function Returning Tree Structure

### DISCUSSION THREAD

**Discussion by dcd-api** [mermaid-js/mermaid#4414] [Ideas]
Hello Mermaid Team,

I appreciate your work on Mermaid. I'm proposing an enhancement: the possibility of using the Mermaid parser separately from the graph rendering.

This would involve parsing the Mermaid graph definition and returning a tree-like data structure instead of a graph. This could allow more flexibility for graph manipulation, transformation, and different forms of visualization within an application.

I understand this may involve substantial work but I believe it could significantly expand Mermaid's utility. I look forward to hearing your thoughts.

**Comment by dcd-api** [comment]
I want to emphasize that my intention to sponsor this project is contingent upon the implementation of the suggested feature. Once the independent parser functionality, returning a tree data structure, is implemented, I will be more than happy to sponsor this project on a monthly basis. I believe this enhancement will greatly benefit the Mermaid community and I'm excited to contribute to its development.

**Comment by Yokozuna59** [comment]
I'm building one here: [Yokozuna59/mermaid-lint](https://github.com/Yokozuna59/mermaid-lint). You could help me if you want; any help is appreciated! See #4401 for more information.

I'm almost done with parsing `title`, `accTitle`,  and `accDescr`. Although some weird use cases aren't handled yet as they are commented [here](https://github.com/Yokozuna59/mermaid-lint/blob/f7f344c2e655b1f88ebb840914881b51f10fad12/packages/mermaid-parser/tests/common/accessible.test.ts).

**Comment by stereobooster** [comment]
Related https://github.com/mermaid-js/mermaid/pull/5892
